File: problems/leetcode/150-interview/11-Container-With-Most-Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int ans = 0;
        int leftBorder = 0;
        int rightBorder = n - 1;
        while (leftBorder < rightBorder) {
            ans = max(ans, (rightBorder - leftBorder) * min(height[leftBorder], height[rightBorder]));
            if (height[leftBorder] < height[rightBorder]) leftBorder++;
            else rightBorder--;
        }
        return ans;
    }
};

File: problems/leetcode/LeetCode-75/11-Container-With-Most-Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        ans, l, r =0, 0,  len(height)-1
        while l<r:
            ans= max(ans,(r-l)*min(height[l],height[r]))
            if height[l]<height[r]:
                l+=1
            else:
                r-=1
        return ans 
